buda/entities/signin.proto

```diff
@@ -0,0 +1,47 @@
+syntax = "proto3";
+
+package buda;
+
+import "buda/entities/uuid.proto";
+import "google/protobuf/timestamp.proto";
+import "buda/entities/product.proto";
+
+message Signin {
+    enum SigninOption {
+      EMAIL = 0;
+      FACEBOOK = 1;
+      TWITTER = 2;
+      LINKEDIN = 3;
+      GOOGLE = 4;
+    }
+
+    enum SigninClient {
+        WEB = 0;
+        PUBLISH = 1;
+        ANALYZE = 3;
+        IPHONE = 4;
+        ANDROID = 5;
+        OTHER = 6;
+    }
+
+
+    message UserAgent {
+        string browser = 1;
+        string version = 2;
+        string os = 3;
+        string device = 4;
+      }
+
+    Uuid id = 1;
+    Uuid user_id = 2;
+    Uuid visitor_id = 3;
+    Uuid legacy_visitor_id = 100; //buffer-web visitor_id, will be deprecated
+    google.protobuf.Timestamp created_at = 4;
+    SigninClient client = 5;
+    SigninOption option = 6;
+    Product product = 7;
+    UserAgent user_agent = 8;
+    Uuid client_id = 9;
```

Comment:
Client id should be a string representation of a mongo objectid, not necessarily a uuid as per standards. Does that make a difference on the data pipeline side?

[uuid v4](https://www.uuidgenerator.net/), 36 characters, ex. `5a5775f3-7dff-4800-9637-903b7aba8d99`
[Mongo's ObjectId](https://docs.mongodb.com/manual/reference/method/ObjectId/), 24 characters, ex. `507f191e810c19729de860ea `